include/game/Coordinates.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <utility>
+
+class Coordinates {
+public:
+    struct CoordX {
+        explicit constexpr CoordX(int coordX)
+            : value(coordX) {}
+
+        constexpr bool operator==(const CoordX& other) const {
+            return value == other.value;
+        }
+
+        int value{0};
+    };
+
+    struct CoordY {
```

Comment:
Duplication.

Reply:
Consider template for axis to eliminate duplication:
    Coordinates(Coordinate<Axis::X> posX, Coordinate<Axis::Y> posY)

Reply:
I'm not sure how to implement that.

Reply:
Refactored